ow_faults_detection/src/FaultDetector.cpp

```diff
@@ -59,52 +59,20 @@ void FaultDetector::setFaultsMessageHeader(fault_msg& msg)
   msg.header.frame_id = "world";
 }
 
-// publish system messages
-void FaultDetector::publishSystemFaultsMessage()
+template<typename faults_msg_pub_t, typename faults_msg_t, typename faults_flags_t>
+void FaultDetector::publishFaultsMessage(faults_msg_pub_t& faults_msg_pub, faults_msg_t& faults_msg, faults_flags_t faults_flags)
```

Comment:
Readability suggestion: the parameter names could be a lot shorter and simpler, e.g. `pub or publisher, msg or message, flags`.  And the line should be shortened in any case -- an easy way is to stack the parameters.